envs/monkey_zoo/blackbox/log_handlers/monkey_logs_downloader.py

```diff
@@ -1,25 +1,63 @@
 import logging
+from pathlib import Path
+from threading import Thread
+from typing import List, Mapping
 
-from envs.monkey_zoo.blackbox.log_handlers.monkey_log import MonkeyLog
+from common.types import MachineID
+from monkey_island.cc.models import Agent, Machine
 
 LOGGER = logging.getLogger(__name__)
 
 
 class MonkeyLogsDownloader(object):
     def __init__(self, island_client, log_dir_path):
         self.island_client = island_client
-        self.log_dir_path = log_dir_path
-        self.monkey_log_paths = []
+        self.log_dir_path = Path(log_dir_path)
+        self.monkey_log_paths: List[Path] = []
 
     def download_monkey_logs(self):
-        LOGGER.info("Downloading each monkey log.")
-        all_monkeys = self.island_client.get_all_monkeys_from_db()
-        for monkey in all_monkeys:
-            downloaded_log_path = self._download_monkey_log(monkey)
-            if downloaded_log_path:
-                self.monkey_log_paths.append(downloaded_log_path)
-
-    def _download_monkey_log(self, monkey):
-        log_handler = MonkeyLog(monkey, self.log_dir_path)
-        download_successful = log_handler.download_log(self.island_client)
-        return log_handler.get_log_path_for_monkey(monkey) if download_successful else None
+        try:
+            LOGGER.info("Downloading each monkey log.")
+
+            agents = self.island_client.get_agents()
+            machines = self.island_client.get_machines()
+
+            download_threads: List[Thread] = []
+
+            # TODO: Does downloading logs concurrently still improve performance after resolving
+            #       https://github.com/guardicore/monkey/issues/2383?
```

Comment:
Performance should be negligible in this case

Reply:
It's not negiligible. It drops the time to download logs from 11.5 seconds to 6.5 seconds. Fixing #2383 drops it to 1.5 seconds.  Multiple by 8 tests.